pkg/forwarder/domain_forwarder.go

```diff
@@ -188,6 +213,9 @@ func (f *domainForwarder) Stop(purgeHighPrio bool) {
 		return
 	}
 
+	if f.connectionResetInterval != 0 {
+		f.stopConnectionReset <- true
+	}
```

Comment:
Should we wait for the Connection reset loop to exit before stopping the workers ?

Reply:
I think this would be ideal, but I also don't think it's a big deal.

Reply:
I think it would add complexity for almost nothing (for the same reason that we don't do that when stopping the goroutine running `handleFailedTransactions`): this channel is unbuffered so it will block until the listening goroutine actually handles the message, and then all we care about is that the listening goroutine stops eventually.